src/modules/languages/clojure.nix

```diff
@@ -6,12 +6,26 @@ in
 {
   options.languages.clojure = {
     enable = lib.mkEnableOption "tools for Clojure development";
+    jdk.package = lib.mkOption {
+      type = lib.types.package;
+      example = lib.literalExpression "pkgs.jdk8";
+      default = pkgs.jdk;
+      defaultText = lib.literalExpression "pkgs.jdk";
+      description = ''
+        The JDK package to use.
+        This will also become available as `JAVA_HOME`.
+      '';
+    };
   };
 
   config = lib.mkIf cfg.enable {
     packages = with pkgs; [
-      clojure
+      (clojure.override {
+        jdk = cfg.jdk.package;
+      })
       clojure-lsp
     ];
+
+    env.JAVA_HOME = cfg.jdk.package.home;
```

Comment:
I wonder if we should instead set `languages.java.enable = true;` and then use `config.languages.java.package`?

Reply:
Yeah, I think that's better. Changed it in the latest commit.

Reply:
java already sets `JAVA_HOME`